unified_planning/model/ma_problem.py

```diff
@@ -0,0 +1,683 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""This module defines the MultiAgentProblem class."""
+
+import sys
+import copy
+from unified_planning.shortcuts import *
+import unified_planning.model.operators as op
+from unified_planning.model.types import domain_size, domain_item
+from unified_planning.environment import get_env, Environment
+from unified_planning.model.walkers import OperatorsExtractor
+from unified_planning.model.fluent import Fluent
+from unified_planning.model.action import *
+from fractions import Fraction
+from typing import List, Dict, Set, Union, Optional
+from unified_planning.exceptions import (
+    UPProblemDefinitionError,
+    UPTypeError,
+    UPValueError,
+    UPExpressionDefinitionError,
+)
+from unified_planning.model.agent import Agent
+from unified_planning.io.pddl_writer import PDDLWriter
+from unified_planning.model.environment_ma import Environment_ma
+import collections
+from typing import List, Union, cast
+from unified_planning.io.pddl_writer_ma import PDDLWriter_MA
+from unified_planning.model.walkers.substituter import Substituter
+import re
+import os
+from typing import IO
+from io import StringIO
+
+from unified_planning.model.abstract_problem import AbstractProblem
+from unified_planning.model.mixins import (
+    ActionsSetMixin,
+    FluentsSetMixin,
+    ObjectsSetMixin,
+    UserTypesSetMixin,
+    AgentsSetMixin,
+)
+
+
+class MultiAgentProblem(
+    # Problem,
+    AbstractProblem,
+    UserTypesSetMixin,
+    FluentsSetMixin,
+    ActionsSetMixin,
+    ObjectsSetMixin,
+    AgentsSetMixin,
+):
+    """Represents a planning MultiAgentProblem."""
+
+    def __init__(
+        self,
+        name: str = None,
+        env: "up.environment.Environment" = None,
+        env_ma: Union["up.model.environment_ma.Environment_ma", None] = None,
```

Comment:
This should not be needed at all given that the `MAProblem` is already a `FluentSetMixin`

Reply:
No, MAEnvironment will survive (see comment above)